src/ppx/scene/scene_mesh.cpp

```diff
@@ -0,0 +1,142 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ppx/scene/scene_mesh.h"
+#include "ppx/grfx/grfx_device.h"
+
+namespace ppx {
+namespace scene {
+
+// -------------------------------------------------------------------------------------------------
+// MeshData
+// -------------------------------------------------------------------------------------------------
+MeshData::MeshData(
+    const scene::VertexAttributeFlags& availableVertexAttributes,
+    grfx::Mesh*                        pGpuMesh)
```

Comment:
It's because of the destroy pattern that we have in grfx::Device. It's still heavily raw pointer based because there was a desire to have on demand destroy (because of Vulkan). There was a larger discussion earlier in the PR regarding if we should change that to something like shared_ptr.